### PR TITLE
Cleanup usage of `unwrap()` in `re_data_store`

### PR DESCRIPTION
**Pull request by Artxiom**
### What

Part of #6330
Removes or explicitly allows unwrap() where it makes sense.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6334?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6334?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6334)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -428,6 +428,7 @@ verbose_file_reads = "warn"
 wildcard_dependencies = "warn"
 wildcard_imports = "warn"
 zero_sized_map_values = "warn"
+unwrap_used = "warn"
```
Comment (emilk) [inline Cargo.toml:431]: please sort list

File: crates/re_data_store/src/store_read.rs [hunk 2]
```diff
@@ -454,7 +454,9 @@ impl IndexedTable {
     pub fn find_bucket(&self, time: TimeInt) -> (TimeInt, &IndexedBucket) {
         // This cannot fail, `iter_bucket` is guaranteed to always yield at least one bucket,
         // since indexed tables always spawn with a default bucket that covers [-∞;+∞].
-        self.range_buckets_rev(..=time).next().unwrap()
+        self.range_buckets_rev(..=time)
+            .next()
+            .expect("this cannot fail")
```
Comment (emilk) [inline crates/re_data_store/src/store_read.rs:459]: Better to use `#[allow(clippy::unwrap_used)]` for things like these imho (slightly less code generated)

**Review by emilk — APPROVED** [review]

**Comment by Artxiom** [comment]
@emilk As I don't have write access - how do you guys handle usually external contributions? Do I need to check the other boxes or do anything else, or will you take over from here?

**Comment by emilk** [comment]
Our CI for external contributors is slightly broken: https://github.com/rerun-io/rerun/issues/1994

The checkboxes you have control over though

**Comment by Artxiom** [comment]
Haha, okay.

I just meant what are the guideline for the checkboxes? The first two are obvious but when is it necessary to e.g. test the web demos or when to add something to the checklist, what's a good PR title, etc? And I can't change the labels anyway.
So should I always just leave the remaining checkboxes unchecked or just check them all?

And thanks!

**Pull request merged**